### PR TITLE
Feat/fee 516 timezone enhancements

### PR DESCRIPTION
**Pull request by apryka**
https://metakeen.atlassian.net/browse/FEE-516

### CODE REVIEW

File: packages/ui-core/src/components/timezone/timezone.component.tsx [hunk 1]
```diff
@@ -1,74 +1,136 @@
...
+  onChange: (timezone: TimezoneType) => void;
+};
+
+type RowProps = {
+  data: Options;
+  index: number;
+  style: Record<string, any>;
+};
+
+const Row: FC<RowProps> = ({ data, index, style }) => {
```
Comment (maciejrybaniec) [inline packages/ui-core/src/components/timezone/timezone.component.tsx:44]: Please, extract it to separate file under `components` directory.

File: packages/ui-core/src/components/timezone/timezone.component.tsx [hunk 2]
```diff
@@ -1,74 +1,136 @@
...
-        </Dropdown>
+        <AnimatePresence>
+          {isOpen && (
+            <CustomDropdown
+              position={dropdownPosition}
+              {...dropdownMotion[dropdownPosition]}
+            >
+              <ReactWindowList
+                ref={listRef}
+                height={150}
```
Comment (maciejrybaniec) [inline packages/ui-core/src/components/timezone/timezone.component.tsx:122]: Please, move it as component property `listHeight` with default value set to `150`

File: packages/ui-core/src/components/timezone/timezone.component.tsx [hunk 3]
```diff
@@ -1,74 +1,136 @@
...
-import { TIMEZONES } from './constants';
+import { Timezone as TimezoneType, DropdownPosition, Options } from './types';
+import { dropdownMotion } from './motion';
 
 type Props = {
+  /** Timezones */
+  timezones: TimezoneType[];
   /** Timezone value */
-  timezone?: Timezones;
+  timezone?: TimezoneType;
```
Comment (maciejrybaniec) [inline packages/ui-core/src/components/timezone/timezone.component.tsx:25]: In my opinion we should not stick `1:1` with back-end schema. Current `timezone` value should be only `string` - that represents named timezone. Providing **utcOffset** just to set value should not be required a Explorer / Creator will be forced to persist it somewhere.

**Review by mczapkowicz — APPROVED** [review]

File: packages/ui-core/src/components/timezone/timezone.component.tsx [hunk 4]
```diff
@@ -1,74 +1,136 @@
...
-    []
-  );
+  const listRef = useRef(null);
+
+  const options: Options = {
+    timezones,
+    timezone,
+    onChange,
+  };
+
```
Comment (maciejrybaniec) [inline packages/ui-core/src/components/timezone/timezone.component.tsx:78]:
```
const { name, offset } = useMemo(() => {
if (timezones.length === 0) {
  return {
   name: timezone
   offset: null,
  };
}
}, [timezone, timezones]) 
```

**Review by maciejrybaniec — CHANGES_REQUESTED** [review]

File: packages/ui-core/src/components/timezone/timezone.component.tsx [hunk 5]
```diff
@@ -1,74 +1,136 @@
...
+  DropableContent,
+} from './timezone.styles';
 
-import { TIMEZONES } from './constants';
+import { Timezone as TimezoneType, DropdownPosition, Options } from './types';
+import { dropdownMotion } from './motion';
 
 type Props = {
+  /** Timezones */
+  timezones: TimezoneType[];
```
Comment (maciejrybaniec) [inline packages/ui-core/src/components/timezone/timezone.component.tsx:23]: `TimezoneType` -> `TimezoneItem`

**Review by maciejrybaniec — APPROVED** [review]

**Pull request merged**